pkg/s3/middleware/drop_accept_encoding.go

```diff
@@ -53,6 +55,8 @@ func DropAcceptEncoding(o *s3.Options) {
 
 // dropAcceptEncodingHeader is a middleware function that removes the Accept-Encoding header from the request.
 // This is necessary for compatibility with certain S3-like storage providers.
+//
+//nolint:lll
```

Comment:
Can we just split this across multiple lines rather than turning off the lint?

Reply:
We can. I tried before adding it but obviously broke the line at the wrong point and Go was particularly fussy about the line-break.  It does work though when you break after the opening ( of the function parameters.  Also replaced the long stackoverflow link with a short version